src/stateManagement/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Legacy state management
```

Comment:
In what way is this legacy? Because it doesn't use the store? Its all widely used for now, so I feel "Legacy" might be a bit confusing.

Reply:
Good call-out. It's not legacy yet. I was a bit hurried trying to shell out docs, and my brain can't wait for this to change in the next iteration I guess. I'll update accordingly.